spec-dtslint/observables/forkJoin-spec.ts

```diff
@@ -0,0 +1,92 @@
+import { of, forkJoin } from 'rxjs';
+
+it('should infer correctly with 1 parameter', () => {
+  const a = of(1, 2, 3);
+  const res = forkJoin(a); // $ExpectType Observable<number[]>
+});
+
+it('should infer correctly with 2 parameters', () => {
+  const a = of(1, 2, 3);
+  const b = of('a', 'b', 'c');
+  const res = forkJoin(a, b); // $ExpectType Observable<[number, string]>
+});
+
+it('should infer correctly with 3 parameters', () => {
+  const a = of(1, 2, 3);
+  const b = of('a', 'b', 'c');
+  const c = of(1, 2, 3);
+  const res = forkJoin(a, b, c); // $ExpectType Observable<[number, string, number]>
+});
+
+it('should infer correctly with 4 parameters', () => {
+  const a = of(1, 2, 3);
+  const b = of('a', 'b', 'c');
+  const c = of(1, 2, 3);
+  const d = of(1, 2, 3);
+  const res = forkJoin(a, b, c, d); // $ExpectType Observable<[number, string, number, number]>
+});
+
+it('should infer correctly with 5 parameters', () => {
+  const a = of(1, 2, 3);
+  const b = of('a', 'b', 'c');
+  const c = of(1, 2, 3);
+  const d = of(1, 2, 3);
+  const e = of(1, 2, 3);
+  const res = forkJoin(a, b, c, d, e); // $ExpectType Observable<[number, string, number, number, number]>
+});
+
+it('should infer correctly with 6 parameters', () => {
+  const a = of(1, 2, 3);
+  const b = of('a', 'b', 'c');
+  const c = of(1, 2, 3);
+  const d = of(1, 2, 3);
+  const e = of(1, 2, 3);
+  const f = of(1, 2, 3);
+  const res = forkJoin(a, b, c, d, e, f); // $ExpectType Observable<[number, string, number, number, number, number]>
+});
+
+it('should infer of type any for more than 6 parameters', () => {
+  const a = of(1, 2, 3);
+  const b = of('a', 'b', 'c');
+  const c = of(1, 2, 3);
+  const d = of(1, 2, 3);
+  const e = of(1, 2, 3);
+  const f = of(1, 2, 3);
+  const g = of(1, 2, 3);
+  const res = forkJoin(a, b, c, d, e, f, g); // $ExpectType Observable<any>
+});
+
+it('should infer correctly for array of 1 observable', () => {
+  const a = [of(1, 2, 3)];
+  const res = forkJoin(a); // $ExpectType Observable<number[]>
+});
+
+it('should infer correctly for array of 2 observables', () => {
```

Comment:
This test - and the tests following it - should be commented out. They have exposed a bug that will have to be addressed in a separate PR. This one should be inferring:

```ts
// $ExpectType Observable<[string, number]>
```

The tests describe the current behaviour, but that behaviour is not what's intended and it's not how the implementation behaves.